File: Modules/Default/Governate/GovernateColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace Serene2.Default.Columns;

[ColumnsScript("Default.Governate")]
[BasedOnRow(typeof(GovernateRow), CheckNames = true)]
public class GovernateColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int GovernateId { get; set; }
    [EditLink]
    public string GovernateName { get; set; }
}

File: Modules/ServerTypes/Default/GovernateColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { GovernateRow } from "./GovernateRow";

export interface GovernateColumns {
    GovernateId: Column<GovernateRow>;
    GovernateName: Column<GovernateRow>;
}

export class GovernateColumns extends ColumnsBase<GovernateRow> {
    static readonly columnsKey = 'Default.Governate';
    static readonly Fields = fieldsProxy<GovernateColumns>();
}